osquery/tables/system/windows/programs.cpp

```diff
@@ -88,6 +194,120 @@ void keyEnumPrograms(const std::string& key,
+
+        // btime "Birth Time"
+        // When a file is first created, btime is set and does not change
+        // ::Warning:: not all filesystems support btime
+        // Older file systems such as ext3 or FAT32 will have this missing
+        WINDOWS_STAT file_stat;
+        auto rtn = platformStat(filePath.c_str(), &file_stat);
+        if (rtn.ok()) {
+          result["install_date"] = formatTimestampToDate(file_stat.btime);
+        }
+
+        auto s = osquery::pathExists(filePath);
+        if (!s.ok()) {
+          // Skip this package, as we cannot find the manifest file
+          // We can extract some information from the registry key
+          // <PackageName>_<Version>_<Architecture>__<PublisherHash>
+          // Example:
+          // MSTeams_25060.205.3499.6849_arm64__8wekyb3d8bbwe
+          // But all proper MSIX packages have an AppxManifest.xml file
+          VLOG(1) << "Cannot find manifest file:'" + filePath + "'";
+          result.clear();
+          continue;
+        }
+
+        std::string xmlContent;
+        s = osquery::readFile(filePath, xmlContent);
+        if (!s.ok()) {
+          // Skip this package, as we cannot read the manifest file
+          VLOG(1) << "Cannot read manifest file:'" + filePath + "'";
+          result.clear();
+          continue;
+        }
+
+        // Find the Identity tag, extract attributes
+        std::string identityTag = findSelfClosingTag(xmlContent, "Identity");
+        if (!identityTag.empty()) {
+          auto attributes = parseAttributes(identityTag);
+          result["name"] = attributes["Name"];
+          result["publisher"] = attributes["Publisher"];
+          result["version"] = attributes["Version"];
+        }
+
+        // Find the Properties tag, extract child tags
+        std::string propertiesTag = getTagContent(xmlContent, "Properties");
+        if (!propertiesTag.empty()) {
+          auto displayName = getTagContent(propertiesTag, "DisplayName");
+          auto publisherDisplayName =
+              getTagContent(propertiesTag, "PublisherDisplayName");
+
+          // "ms-resource:" prefix means that the string is dynamically
+          // generated from a .pri file .pri file is a binary index of all
+          // localized and scaled resources compiled from .resw files or
+          // .resources at build time
+          if (!displayName.empty() &&
+              displayName.find("ms-resource") == std::string::npos) {
+            result["name"] = displayName;
+          }
+          if (!publisherDisplayName.empty() &&
+              publisherDisplayName.find("ms-resource") == std::string::npos) {
+            result["publisher"] = publisherDisplayName;
+          }
+        }
+
+        // Done processing this package registry entries
+        // No need to read anymore keys
+        continue;
+      }
+    } // end processing package registry entry
+
+    if (!result.empty()) {
+      auto packageKey = result["package_family_name"];
```

Comment:
There was a question about what I was grouping by and I wanted to explicitly add some context here. 

For MSIX applications Windows enforces that only one version of a package with the same identity can exist per user at a time. `PackageFamilyName` is a combination of `PackageName` + `PublisherHash`. The reason why I don't have version in there was because if you try to install an MSIX application with a version greater it will automatically upgrade and uninstall the older version. If you try to downgrade, it will throw an error.

Reply:
I wonder if that will be true in the future. 🤷 Doesn't feel like a breaking change if we need to update this though.

Reply:
I think if Microsoft decides they will allow more than one version per identify, the most that this would change is the number of rows returned. Which I think in this case would be a "bug" if it didn't return all the versions, so, indeed not a breaking change.